tests/unit/Rules/EachTest.php

```diff
@@ -16,13 +16,52 @@
  * @covers Respect\Validation\Rules\Each
  * @covers Respect\Validation\Exceptions\EachException
  */
-class EachTest extends \PHPUnit_Framework_TestCase
+class EachTest extends RuleTestCase
 {
+    public function providerForValidInput()
+    {
+        $ruleNotEmpty = new Each($this->getRuleMock(true, 'NotEmpty'));
+        $ruleAlphaItemIntKey = new Each($this->getRuleMock(true, 'Alpha'), $this->getRuleMock(true, 'IntVal'));
+        $ruleOnlyKeyValidation = new Each(null, $this->getRuleMock(true, 'IntVal'));
```

Comment:
The names of the rules here is just irrelevante, it's just the name of the mock classes